tests/functional/adapter/materialization/test_materialized_view.py

```diff
@@ -40,3 +47,97 @@ def test_mv_is_dropped_when_model_runs_view(self, project):
         }
         check_relation_types(project.adapter, expected)
+
+
+@pytest.mark.iceberg
+class TestIcebergMaterializedViewCreate:
+    # Configuration in dbt_project.yml
+    @pytest.fixture(scope="class")
+    def project_config_update(self):
+        return {
+            "name": "mv_test",
+            "models": {
+                "+materialized": "materialized_view",
+            },
+            "seeds": {
+                "+column_types": {"some_date": "timestamp(6)"},
+            },
+        }
+
+    # Everything that goes in the "seeds" directory
+    @pytest.fixture(scope="class")
+    def seeds(self):
+        return {
+            "seed.csv": seed_csv,
+        }
+
+    # Everything that goes in the "models" directory
+    @pytest.fixture(scope="class")
+    def models(self):
+        return {
+            "mat_view.sql": model_sql,
+        }
+
+    def test_mv_is_created_and_refreshed(self, project):
+        catalog = project.adapter.config.credentials.database
+        schema = project.adapter.config.credentials.schema
+
+        # Seed seed
+        results = run_dbt(["seed"], expect_pass=True)
+        assert len(results) == 1
+
+        # Create MV
+        results = run_dbt(["run"], expect_pass=True)
+        assert len(results) == 1
+
+        # Check if the data was loaded correctly
+        check_relations_equal(project.adapter, ["seed", "mat_view"])
+
+        # Add one row to seed
+        sql = f"""INSERT INTO {catalog}.{schema}.seed
+        VALUES (5, 'Mateo', timestamp '2014-09-07 17:04:27')""".strip()
+        run_sql_with_adapter(project.adapter, sql, fetch="all")
+
+        # Refresh MV
+        results = run_dbt(["run"], expect_pass=True)
+        assert len(results) == 1
+
+        # Check if one row is added in MV
+        sql = f"""select * from {catalog}.{schema}.mat_view
+        """.strip()
+        results = run_sql_with_adapter(project.adapter, sql, fetch="all")
+        assert len(results) == 5
+
+        # Cleanup
+        project.adapter.execute("DROP MATERIALIZED VIEW mat_view")
+
+    def mv_overrides_relation(self, project, relation_type):
+        # Create relation with same name
+        project.adapter.execute(f"CREATE {relation_type} mat_view AS SELECT 3 c")
+        expected = {"mat_view": relation_type.replace(" ", "")}
+        check_relation_types(project.adapter, expected)
+
+        # Seed seed
+        results = run_dbt(["seed"], expect_pass=True)
+        assert len(results) == 1
+
+        # Create MV, table/view/MV should be dropped beforehand
+        results = run_dbt(["run", "--full-refresh"], expect_pass=True)
+        assert len(results) == 1
+        expected = {"mat_view": "materializedview"}
+        check_relation_types(project.adapter, expected)
+
+        # Check if MV was created correctly
+        check_relations_equal(project.adapter, ["seed", "mat_view"])
+
+        # Cleanup
+        project.adapter.execute("DROP MATERIALIZED VIEW mat_view")
```

Comment:
Is the clean up not automated by dbt. I remember some cleanup scripts being called on the schema.

Reply:
without it I'm receiving:
```
TrinoUserError(type=USER_ERROR, name=ALREADY_EXISTS,
message="Materialized view already exists: test16782701215530683130_test_materialized_view.mat_view", query_id=20230308_100453_04095_5fwa7)
```
on second test run.
So maybe dbt isn't handling MVs cleanup properly?